library/ssl_tls13_client.c

```diff
@@ -422,9 +535,47 @@ int mbedtls_ssl_tls13_handshake_client_step( mbedtls_ssl_context *ssl )
             break;
 
         case MBEDTLS_SSL_SERVER_HELLO:
-            // Stop here : we haven't finished whole flow
-            ret = MBEDTLS_ERR_SSL_FEATURE_UNAVAILABLE;
-            mbedtls_ssl_handshake_set_state( ssl, MBEDTLS_SSL_ENCRYPTED_EXTENSIONS );
+            ret = ssl_tls1_3_read_server_hello( ssl );
+            break;
+
+        case MBEDTLS_SSL_ENCRYPTED_EXTENSIONS:
+            ret = ssl_tls1_3_read_encrypted_extensions( ssl );
```

Comment:
Seems all of the processing functions be renamed as xxx_read_xxx() or xxx_write_xxx(), can this nameing represent the reality?

Reply:
Yes. All state just include one read or write.

Reply:
But seems most of the functions not only do the "read/write", also do the processing. Take the encrypted entensions as one example, it will firstly read data from server side and then parse and process the entensions. The server finished is another example, it firstly ready server finish from server side, then parse and processing, generate kinds of keys etc.  I am not sure whether the simple "xxx_read_xxx()" can cover them.

Reply:
Dose this topic can be closed?

Reply:
Yes.

Reply:
Re-opening this as I think @xkqian has a point here (sorry for coming on this quite late). At https://github.com/ARMmbed/mbedtls/pull/4882#discussion_r695731773 it was agreed about the `write` but regarding `read` I think also that `process` is a better fit to reflect what the handler functions are actually doing. Thus I would prefer to rather go for xxx_process_xxx and xxx_write_xxx.

Reply:
Will add it into description.